services/py-api/src/database/repository/participants_repository.py

```diff
@@ -137,3 +137,29 @@ async def get_verified_random_participants_count(self) -> int:
     async def get_number_registered_teammates(self, team_id: str) -> int:
         """Returns the count of registered participants already in the team."""
         return await self._collection.count_documents({"team_id": ObjectId(team_id)})  # type: ignore
+
+    async def get_verified_random_participants(self) -> Result[List[Participant], Exception]:
+        """Returns a list of verified random participants."""
+
+        try:
+            LOG.info("Fetching all random participants...")
+
+            participants_data = await self._collection.find(filter={"email_verified": True, "team_id": None}).to_list(
+                length=None
+            )
+
+            participants = []
+
+            for doc in participants_data:
+                doc_copy = dict(doc)
+
+                doc_copy["id"] = str(doc_copy.pop("_id"))
```

Comment:
I am still debating whether we need a dict copy. I am not sure if we are going to get "RuntimeError: dictionary changed size during iteration"  error, if we directly do ` doc["id"] = str(doc.pop("_id"))`. This is also valid for everywhere we do deepcopy of dict for this exact purpose.